File: app/src/main/java/nl/rsdt/japp/jotial/maps/clustering/osm/OsmScoutingGroepClusterManager.java
package nl.rsdt.japp.jotial.maps.clustering.osm;

import java.util.ArrayList;
import java.util.Collection;

import nl.rsdt.japp.jotial.data.structures.area348.ScoutingGroepInfo;
import nl.rsdt.japp.jotial.maps.clustering.ClusterManagerInterface;
import nl.rsdt.japp.jotial.maps.wrapper.osm.OsmJotiMap;

/**
 * Created by mattijn on 10/08/17.
 */

public class OsmScoutingGroepClusterManager implements ClusterManagerInterface {

    private final Collection<ScoutingGroepInfo> infos;
    private final OsmMarkerContainer markers;

    public OsmScoutingGroepClusterManager(OsmJotiMap jotiMap) {
        if (!(jotiMap instanceof OsmJotiMap)){
            throw new RuntimeException("this class can only be usd with osm");
        }
        infos = new ArrayList<>();
        markers = new OsmMarkerContainer(jotiMap);
    }

    @Override
    public void addItems(final ArrayList<ScoutingGroepInfo> buffer) {
        for (final ScoutingGroepInfo info : buffer){
            infos.add(info);
            markers.add(info);
        }
        markers.showMarkers();
    }

    @Override
    public void cluster() {

    }

    @Override
    public Collection<ScoutingGroepInfo> getItems() {
        return infos;
    }

    @Override
    public void clearItems() {
        infos.clear();
        markers.clear();
    }
}

File: app/src/main/java/nl/rsdt/japp/jotial/maps/clustering/osm/OsmScoutingGroepClusterManager.kt
package nl.rsdt.japp.jotial.maps.clustering.osm

import nl.rsdt.japp.jotial.data.structures.area348.ScoutingGroepInfo
import nl.rsdt.japp.jotial.maps.clustering.ClusterManagerInterface
import nl.rsdt.japp.jotial.maps.wrapper.osm.OsmJotiMap
import java.util.*

/**
 * Created by mattijn on 10/08/17.
 */

class OsmScoutingGroepClusterManager(jotiMap: OsmJotiMap) : ClusterManagerInterface {

    private val infos: MutableCollection<ScoutingGroepInfo>
    private val markers: OsmMarkerContainer

    override val items: List<ScoutingGroepInfo>
        get() = infos.toList()

    init {
        infos = ArrayList()
        markers = OsmMarkerContainer(jotiMap)
    }

    override fun addItems(buffer: ArrayList<ScoutingGroepInfo>) {
        val infosToAdd = mutableListOf<ScoutingGroepInfo>()
        for (info in buffer) {
            infosToAdd.add(info)
            markers.add(info)
        }
        infos.addAll(infosToAdd)
        markers.showMarkers()
    }

    override fun cluster() {

    }

    override fun clearItems() {
        infos.clear()
        markers.clear()
    }
}
